csharp/src/AddressBook/Addressbook.cs

```diff
@@ -241,7 +236,7 @@ public sealed partial class Person : pb::IMessage<Person> {
             break;
           }
           case 34: {
-            input.ReadMessageArray(phone_, global::Google.Protobuf.Examples.AddressBook.Person.Types.PhoneNumber.Parser);
+            phone_.AddEntriesFrom(input, _repeated_phone_codec);
```

Comment:
is there a reason why phone_ is separate from the _repeated_phone_codec?   the phone_ field will only ever use that specific codec, right?
the mehod signature   AddEntriesFrom(codedInputStream, codec) looks a bit odd - perhaps we could make it just phone_.AddEntriesFrom(input) and phone would know which codec to use? It is completely fine to do this in the future (if ever), I actually don't feel strong about this.